### PR TITLE
Add anniversary support

### PR DESCRIPTION
**Pull request by jcalarcon98**
# Birthday Slack Bot

## Authors

- Jean Carlos Alarcón @jcalarcon98 

## Description

This PR adds Anniversary support to send messages like:

image

### CODE REVIEW

File: src/controllers/anniversary_message.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return template.format(username=username, anniversary_years=anniversary_years)
+
+    @staticmethod
+    def get_anniversary_employees(employees: List[dict], anniversary_field) -> List[dict]:
+        def is_anniversary(anniversary: str):
+            if not anniversary:
+                return False
+            delimiter = '-'
+            current_day_month: tuple = dateutils.get_current_day_month(EnvManager.UTC_HOUR_OFFSET)
+            anniversary_day, anniversary_month, _ = tuple(int(value) for value in reversed(anniversary.split(delimiter)))
```
Comment (Wason1797) [inline src/controllers/anniversary_message.py]: Maybe we can extract this logic to the controller?

Seems very similar to the birthday message selector

File: src/controllers/base.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        characters_to_ignore = '.!?'
+        template = template.replace('\n', ' ').strip(characters_to_ignore).split(' ')
+        searchable_template = [word.strip(characters_to_ignore) for word in template]
+        keyword_count = OrderedDict(zip(cls.gif_keywords, repeat(0, len(cls.gif_keywords))))
+        total_count = 0
+        for word in searchable_template:
+            if (lower_word := word.lower()) in keyword_count:
+                keyword_count[lower_word] += 1
+                total_count += 1
+        return choices(list(keyword_count.keys()), weights=[count / total_count for count in keyword_count.values()])
```
Comment (Wason1797) [inline src/controllers/base.py]: Could we have a way to harden this?

If no keyword is found in the template, probably we can use a default one

```python
if total_count == 0:
  return choice(list(keyword_count.keys()))
```

File: src/utils/dateutils.py [hunk 3]
```diff
@@ -8,3 +8,9 @@ def get_day_month(date: datetime) -> tuple:
 def get_current_day_month(utc_hour_offset: str) -> tuple:
     current_date = datetime.utcnow() + timedelta(hours=int(utc_hour_offset))
     return get_day_month(current_date)
+
+
+def get_years_difference_from_current_date(utc_hour_offset: str, date: str, date_format: str = '%Y-%m-%d') -> int:
+    date = datetime.strptime(date, date_format)
+    current_date = datetime.utcnow() + timedelta(hours=int(utc_hour_offset))
```
Comment (Wason1797) [inline src/utils/dateutils.py:15]: Maybe we should take into consideration the utc-offset here as well

We need to consider that lambdas run in GMT and We are trying to send messages to a  GMT-5 timezone

**Review by Wason1797 — COMMENTED** [review]
Some things to consider

File: src/controllers/anniversary_message.py (hunk 1 above)
Comment (jcalarcon98) [reply src/controllers/anniversary_message.py]: Sure, the function was improved and it makes more sense to be a part of dateutils

File: src/controllers/base.py (hunk 2 above)
Comment (jcalarcon98) [reply src/controllers/base.py]: Perfect Wlady

**Review by jeffqev — APPROVED** [review]

**Pull request merged**